snippets/base/forms.py

```diff
@@ -575,6 +575,32 @@ def clean(self):
                 'There can be only one Main Text variable type per template')
 
 
+class AutoTranslatorWidget(forms.Select):
+    def create_option(self, name, value, label, selected, index, subindex=None, attrs=None):
+        """For each key available in each Locale.translation JSON dictionary, create an
+        `translation-{key}` attribute. The attribute will be used by
+        `autoTranslatorWidget.js` to translate fields on Locale selection.
+
+        """
+        option = super().create_option(name, value, label, selected,
+                                       index, subindex=None, attrs=None)
+        if value:
+            translations = json.loads(
+                models.Locale.objects.get(id=value).translations
+            )
+
+            for key, value in translations.items():
+                option['attrs'][f'translation-{key}'] = value
+
+        return option
+
+    class Media:
+        js = [
+            'js/lib/jquery-3.3.1.min.js',
```

Comment:
Are we including jQuery solely for the `autoTranslatorWidget.js` file? If so, happy to help refactor so we don't need the jQuery dependency.

Reply:
No, we have jQuery for a few things, you can see that it's included in other forms as well. Suggestions to improve the JS part with or without jquery are very welcome though

Reply:
Refactored the code to a simpler version. still makes use of jQuery

Reply:
I think this is looking good. I can't get this running locally [1], so tough to test/write vanilla JS. We can add this JS file to the list if/when we get around to removing jQuery. 😁 

[1] `django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: No module named 'psycopg2'`

Reply:
> [1] `django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: No module named 'psycopg2'`

you probably need to rebuild your docker image

Reply:
Uh, yep, that did it. 🤦‍♂